### PR TITLE
PR:  Use file type icons in the File Switcher

### LINKED ISSUE #8580: Improvements to the File switcher
I've been using VSCode a lot these days and I find its file switcher very neat and well designed:

seleccion_001

in comparison to our own:

seleccion_004

So I'd like to improve it for Spyder 4 with @juanis2112's help. These are my suggestions:

- [ ] Use a single row to show a file name and its path (instead of two, which is what we have right now).
- [ ] Use the same font size for file names as the one used in the symbol switcher (@juanis2112, you can get that one by pressing `Cmd+Alt+P`). Also, use a font size one point smaller than that one for paths.
- [ ] Use a grey color for paths instead of italics.
- [ ] Make the switcher taller (look how many files are shown in VSCode in comparison to our own switcher). We should see at most 15 files on it.
- [ ] Use the newly introduced file type icons to show them to the left of each file name (instead of the generic text file icon).
- [ ] Remove the section header rows we have right now (Editor/Notebook). Instead, show those headers in blue to the right of the first file in the corresponding section. Also show an horizontal line when the section finishes.
- [ ] Add a new section called *Recent files* to the switcher that shows the last 5 or 10 opened files that are not present in the Editor.

@juanis2112, please address each of these improvements in a different pull request.

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

I implemented the file type icons in the file switcher.

screen shot 2019-01-31 at 10 37 28 pm

### Issue(s) Resolved

Fixes item 5 of #8580 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Juanis2112

### CODE REVIEW

File: spyder/plugins/explorer/widgets.py [hunk 1]
```diff
@@ -129,68 +129,23 @@ class IconProvider(QFileIconProvider):
     OFFICE_FILES = {'.xlsx': 'ExcelFileIcon', '.docx': 'WordFileIcon',
                     '.pptx': 'PowerpointFileIcon'}
 
+
```
Comment (dalthviz) [inline spyder/plugins/explorer/widgets.py]: @juanis2112 the constants above are still needed?, if not please remove them :)

File: spyder/utils/icon_manager.py [hunk 2]
```diff
@@ -15,6 +16,7 @@
 from spyder.config.base import get_image_path
 from spyder.config.main import CONF
 from spyder.config.gui import is_dark_interface
+from spyder.py3compat import (to_text_string)
```
Comment (dalthviz) [inline spyder/utils/icon_manager.py]: Please remove the parentheses, they are not necessary

File: spyder/utils/icon_manager.py [hunk 3]
```diff
@@ -23,7 +25,20 @@
 else:
     MAIN_FG_COLOR = 'black'
 
-
+BIN_FILES = {x: 'ArchiveFileIcon' for x in ['zip', 'x-tar',
+             'x-7z-compressed', 'rar']}
+DOCUMENT_FILES = {'vnd.ms-powerpoint': 'PowerpointFileIcon',
```
Comment (dalthviz) [inline spyder/utils/icon_manager.py]: Please add a blank line between the dictionaries's variables

**Comment by CAM-Gerlach** [comment]
FYI @juanis2112 great work on this PR but if you write ``Fixes #NNNN`` in a PR post, the PR will close the whole issue, which is presumably not what you intended since it only fixes a part of it.

File: spyder/utils/icon_manager.py [hunk 4]
```diff
@@ -366,3 +381,56 @@ def icon(name, resample=False, icon_path=None):
...
+def get_icon_by_extension(fname):
+    application_icons = {}
+    application_icons.update(BIN_FILES)
+    application_icons.update(DOCUMENT_FILES)
+    if osp.isdir(fname):
+        return icon('DirOpenIcon')
+    else:
+        basename = osp.basename(fname)
+        _, extension = osp.splitext(basename.lower())
+        mime_type, _ = mime.guess_type(basename)
```
[the remark this answers is not in the document]
Comment (dalthviz) [reply spyder/utils/icon_manager.py]: @juanis2112 this is missing :)

File: spyder/utils/icon_manager.py [hunk 5]
```diff
@@ -366,3 +384,56 @@ def icon(name, resample=False, icon_path=None):
             if osp.isfile(icon_path):
                 icon = QIcon(icon_path)
         return icon if icon is not None else QIcon()
+
+
+def get_icon_by_extension(fname):
```
Comment (ccordoba12) [inline spyder/utils/icon_manager.py]: I like this refactoring!! It was a really good idea to have all of this grouped under a single function.

Nicely done @juanis2112! (or was it @dalthviz idea?)

Comment (dalthviz) [reply spyder/utils/icon_manager.py]: I would say it was a team effort :+1:

Comment (ccordoba12) [reply spyder/utils/icon_manager.py]: Great to hear!

File: spyder/utils/icon_manager.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (juanis2112) [reply spyder/utils/icon_manager.py]: @dalthviz done!

**Comment by dalthviz** [comment]
@juanis2112 please fix the conflicts to merge this one, thanks :)

**Pull request merged**